tests/unit/amo/components/TestCollection.js

```diff
@@ -1268,6 +1268,66 @@ describe(__filename, () => {
         username: detail.author.username,
       }),
     );
+
+    sinon.assert.notCalled(root.instance().props.history.push);
+  });
+
+  it('updates the page when removeAddon removes the last addon on the page', () => {
+    const history = createFakeHistory();
+    const authorUserId = 11;
+    const { store } = dispatchSignInActions({ userId: authorUserId });
+
+    const addons = createFakeCollectionAddons();
+    const addonId = addons[0].addon.id;
+    const detail = createFakeCollectionDetail({
+      authorId: authorUserId,
+      // This will simulate only 1 item on the 2nd page.
+      count: DEFAULT_API_PAGE_SIZE + 1,
```

Comment:
We probably need a test with `DEFAULT_API_PAGE_SIZE + 2`.

Reply:
I thought about this but the original case I guess assumes and does this really. Maybe it would be okay to just update that case to include this info so it's clear?

Reply:
That should probably be covered to distinguish the happy path (page = 1).